File: src/backend/TrainingManagement.Api/TrainingManagement.Api/Entities/TrainingRequest.cs
using System;

namespace TrainingManagement.Api.Entities
{
    public class TrainingRequest
    {
        public int Id { get; set; }
        public int EmployeeId { set; get; }
        public int CourseId { get; set; }
        public string RequestReason { get; set; }
        public string Status { get; set; }
        public int? DeptApproverId { get; set; }
        public DateTime? DeptApproveTime { get; set; }
        public string DeptApproveComment { get; set; }
        public int? HrApproverId { get; set; }
        public DateTime? HrFileTime { get; set; }
        public DateTime CreateTime { get; set; }
    }
}

File: src/frontend/src/api/mappers/training-request.ts
import { fromEnvelopeFailure } from '@/api/error'
import type { ApiEnvelopeDto, PageDto, TrainingRequestDto } from '@/api/transport'
import type { DomainPage } from '@/domains/shared'
import type { TrainingRequest, TrainingRequestStatus } from '@/domains/training-request'

const knownStatuses = new Set<TrainingRequestStatus>([
  'PENDING',
  'DEPT_APPROVED',
  'DEPT_REJECTED',
  'HR_FILED',
])

const mapStatus = (value: string | null | undefined): TrainingRequestStatus =>
  knownStatuses.has(value as TrainingRequestStatus) ? (value as TrainingRequestStatus) : 'UNKNOWN'

export function mapTrainingRequest(dto: TrainingRequestDto): TrainingRequest {
  return {
    id: String(dto.id),
    courseId: String(dto.courseId),
    courseName: dto.courseName || '未命名课程',
    employeeId: dto.employeeId,
    employeeName: dto.employeeName || '—',
    departmentName: '—',
    departmentId: dto.deptId == null ? null : String(dto.deptId),
    reason: dto.requestReason || '',
    status: mapStatus(dto.status),
    submittedAt: dto.createTime || '',
    updatedAt: null,
    departmentOpinion: dto.deptApproveComment ?? null,
    hrOpinion: dto.hrFileComment ?? null,
  }
}

export function mapTrainingRequestPage(
  envelope: ApiEnvelopeDto<PageDto<TrainingRequestDto>>,
): DomainPage<TrainingRequest> {
  if (!envelope.success || !envelope.data) throw fromEnvelopeFailure(envelope)
  return { ...envelope.data, items: envelope.data.items.map(mapTrainingRequest) }
}
